gensrc/script/functions.py

```diff
@@ -295,8 +295,8 @@
      'StringFunctions::regexp_extract_prepare', 'StringFunctions::regexp_close'],
     [30330, 'regexp_replace', 'VARCHAR', ['VARCHAR', 'VARCHAR', 'VARCHAR'], 'StringFunctions::regexp_replace',
      'StringFunctions::regexp_replace_prepare', 'StringFunctions::regexp_close'],
-    [30331, 'replace', 'VARCHAR', ['VARCHAR', 'VARCHAR', 'VARCHAR'], 'StringFunctions::regexp_replace',
```

Comment:
For compatibility purpose,  both old and new implementation should retain, in rolling upgradation of cluster, old FE would send old version function two new BE.

Reply:
I get your point. But `replace` was an undocumented API, and has a flawed semantics as an alias to `regexp_replace`.

Instead, I would state in the doc that the correctly implemented `replace` is since 3.1 (3.0 if we are ok for backporting), and for the old behavior use `regexp_replace` instead.

What do you think? @satanson

Reply:
Sth like this? @satanson 
```
     [30332, 'replace', 'VARCHAR', ['VARCHAR', 'VARCHAR', 'VARCHAR'], 'StringFunctions::replace',
      'StringFunctions::replace_prepare', 'StringFunctions::replace_close'],
     # @Deprecated: it will be deleted in the future version, keep it just for compatible
     [30331, 'replace', 'VARCHAR', ['VARCHAR', 'VARCHAR', 'VARCHAR'], 'StringFunctions::regexp_replace',
      'StringFunctions::regexp_replace_prepare', 'StringFunctions::regexp_close'],

```

Reply:
@zuyu yes, like this